model/src/main/proto/exploration.proto

```diff
@@ -0,0 +1,145 @@
+  map<string, VoiceoverMapping> recorded_voiceovers = 1;
+  SubtitledHtml content = 2;
+  // Mapping from content_id to a TranslationMapping
+  map<string, TranslationMapping>written_translations = 3;
+  repeated ParamChange param_changes = 4;
+  string classifier_model_id = 5;
+  Interaction interaction = 6;
+  bool solicit_answer_details = 7;
+}
+
+// Data for customization args.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeCustomizationArgs.kt
+message CustomizationArgs {
+  bool parse_with_jinja = 1;
+  string value = 2;
+}
+
+// Mapping from language_code to a VoiceOver
+message VoiceoverMapping {
+  map<string, Voiceover> voiceover_mapping = 1;
+}
+
+// Data for a single voiceover
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeVoiceover.kt
+message Voiceover {
+  int64 file_size_bytes = 1;
+  bool needs_update = 2;
+  string file_name = 3;
+}
+
+// Data for a single interaction
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeInteractionInstance.kt
+message Interaction {
+  string id = 1;
+  repeated AnswerGroup answer_groups = 2;
+  Solution solution = 3;
+  repeated AnswerGroup confirmed_unclassified_answers = 4;
+  Hint hint = 5;
+  Outcome default_outcome = 6;
+  map<string, CustomizationArgs> customization_args = 7;
+}
+
+// Mapping from language_code to Translation
+message TranslationMapping {
+  map<string, Translation> translation_mapping = 1;
+}
+
+// Data for a single translation
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeWrittenTranslation.kt
+message Translation {
+  string html = 1;
+  bool needs_update = 2;
+}
+
+// Data for a single answer group
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeAnswerGroup.kt
+message AnswerGroup {
+  string tagged_skill_misconception_id = 1;
+  Outcome outcome = 2;
+  repeated RuleSpec rule_specs = 3;
+  repeated TrainingData training_data = 4;
+}
+
+// Data for a single training data instance
+message TrainingData {
+  int32 answer_group_index = 1;
+  repeated string answers = 2;
+}
+
+// Data for a single solution
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeSolution.kt
+message Solution {
+  string interaction_id = 1;
+  bool answer_is_exclusive = 2;
+  string correct_answer = 3;
+  SubtitledHtml explanation = 4;
+}
+
+// Data for a single hint
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeHint.kt
+message Hint {
+  SubtitledHtml hint_content = 1;
+}
+
+// Data for a single outcome
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeOutcome.kt
+message Outcome {
+  string dest = 1;
+  string refresher_exploration_id = 2;
+  SubtitledHtml feedback = 3;
+  repeated ParamChange param_changes = 4;
+  string missing_prerequisite_skill_id = 5;
+  bool labelled_as_correct = 6;
+}
+
+// Data for a single rule spec
+message RuleSpec {
+  map<string, HtmlStringList> inputs = 1;
```

Comment:
Is HtmlStringList the only type for inputs, or can it be something else? It appears this was represented with 'Any' in the original structure.

Reply:
Looking at the file the comment - https://github.com/oppia/oppia/blob/ec254cfcc87843c36e24d68d335cc811536fb2d4/core/domain/state_domain.py#L1146 points to, I believe they are a list of html strings

Reply:
Per https://github.com/oppia/oppia/blob/develop/extensions/interactions/rule_templates.json I think the inputs are typed based on the rule spec. E.g. some can be CodeString, others Fraction, or list of HTML strings. I think for the interactions we're supporting (#28, #29, #30, #31, #32, #33, #34, #35) we should support the corresponding rulespecs for them. Should we be using 'anyof' here, instead?

Reply:
I have added them as part of the InteractionObject message. Added the new types of input based on interactions mentioned in #28 -> #35